cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -1128,6 +1128,46 @@ impl Node<Option<cst::Member>> {
         }
     }
 
+    /// Construct an attribute access or method call on an expression. This also
+    /// handles function calls, but a function call of an arbitrary expression
+    /// is always an error.
+    fn build_expr_accessor<'a>(
+        &self,
+        head: ast::Expr,
+        next: &mut AstAccessor,
+        tail: &'a mut [AstAccessor],
+    ) -> Result<(ast::Expr, &'a mut [AstAccessor])> {
+        use AstAccessor::*;
+        match (next, tail) {
+            // trying to "call" an expression as a function like `(1 + 1)("foo")`. Always an error.
+            (Call(_), _) => Err(self.to_ast_err(ToASTErrorKind::ExpressionCall).into()),
+
+            // method call on arbitrary expression like `[].contains(1)`
+            (Field(id), [Call(args), rest @ ..]) => {
+                // move the expr and args out of the slice
+                let args = std::mem::take(args);
```

Comment:
is this necessary? can we just use `args` directly in the below?

Reply:
Should be avoidable, but `tail` is a slice, and `to_meth` wants to take ownership, so not trivially. I think it would mean rewriting this to work of an `impl Iterator<Item = AstAccessor>`, but then doing the lookahead is kind of annoying.

Reply:
Ah I see, I missed the types involved here.